fs/dafsclient/rmtclient.cpp

```diff
@@ -126,32 +126,44 @@ static class _securitySettings
 
 static CriticalSection              secureContextCrit;
 static Owned<ISecureSocketContext>  secureContextClient;
+static std::unordered_map<std::string, Owned<ISecureSocketContext>> secureCtxClientIssuerMap;
```

Comment:
This caching of issuer based SecureSocketContexts is something to consider commoning up and using elsewhere.  But if we did that, we'd have to insure consistency.  For example don't want to mingle disableClientCertVerification contexts with non disableClientCertVerification contexts.

Reply:
This is only on the client side at the moment, whereas the disableClientCertVerification option is on the server side.
Not sure there's much need to cache on the server side typically, as it will have 1 context, whereas clients are potentially connecting to disparate servers with different certificates.

But as you say, there are prob. other contexts where it would be good to cache.
If more generalized, it should also cope with configuration changes, e.g. on a helm update, which would potentially change tls info etc., and therefore should invalidate the cache.

In fact, I'll look at adding that change, to clear the map if a configuration chagnes, as it may be a gotcha.